src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/StartupProjectRegistrar.cs

```diff
@@ -111,8 +113,8 @@ protected override Task DisposeCoreAsync(bool initialized)
 
         private async Task AddOrRemoveProjectFromStartupProjectList(bool initialize = false)
         {
+            bool isDebuggable = await IsDebuggable().ConfigureAwait(false);
```

Comment:
@davkean mentioned that one only thread is guaranteed to be in the call back at a time. Seems that guarantee is used in other part of our code as well. So should be fine.